Source/WebCore/css/CSSStyleSheet.cpp

```diff
@@ -571,12 +571,18 @@ CSSStyleSheet::RuleMutationScope::RuleMutationScope(CSSStyleSheet* sheet, RuleMu
     m_contentsWereClonedForMutation = m_styleSheet->willMutateRules();
 }
 
+static String modifiedKeyframesRuleName(CSSRule* rule)
+{
+    auto* cssKeyframeRule = dynamicDowncast<CSSKeyframesRule>(rule);
+    return cssKeyframeRule ? cssKeyframeRule->name() : emptyAtom();
+}
+
 CSSStyleSheet::RuleMutationScope::RuleMutationScope(CSSRule* rule)
     : m_styleSheet(rule ? rule->parentStyleSheet() : nullptr)
     , m_mutationType(is<CSSKeyframesRule>(rule) ? KeyframesRuleMutation : OtherMutation)
     , m_contentsWereClonedForMutation(ContentsWereNotClonedForMutation)
     , m_insertedKeyframesRule(nullptr)
-    , m_modifiedKeyframesRuleName(is<CSSKeyframesRule>(rule) ? downcast<CSSKeyframesRule>(*rule).name() : emptyAtom())
+    , m_modifiedKeyframesRuleName(modifiedKeyframesRuleName(rule))
```

Comment:
I think a lambda is more elegant here.

Reply:
A multi line lambda in the middle of the members initialization? Not convinced this will look better than my static function?

Reply:
I made the change, not sure if it's better but why not.

Reply:
It lets you see what’s going on at a glance. Without having to look for the method. Though here, it’s right next to it.